pe.c

```diff
@@ -772,8 +899,18 @@ handle_image (void *data, unsigned int datasize,
 			clear_attrs &= ~MEM_ATTR_X;
 		}
 		update_mem_attrs(addr, length, set_attrs, clear_attrs);
-	}
 
+		if ((Section->Characteristics & EFI_IMAGE_SCN_CNT_CODE ||
+		     Section->Characteristics & EFI_IMAGE_SCN_CNT_INITIALIZED_DATA) &&
+		    Section->Characteristics & EFI_IMAGE_SCN_MEM_READ) {
+			efi_status = cache_section(li, Section->Name, base, raw_length);
+			if (EFI_ERROR(efi_status)) {
+				perror(L"Failed to cache section details\n");
+				BS->FreePages(*alloc_address, *alloc_pages);
+				return efi_status;
+			}
```

Comment:
A debug print here noting a successful caching would be nice - it did help me debug these small issues, something like:

```
dprint(L"Cached section %d (%a) at 0x%016llx, size 0x%016llx\n",
       i, Section->Name,
       (unsigned long long)(uintptr_t)base,
       (unsigned long long)raw_length);
```

Reply:
I've added this in my current tree.